tensorflow/contrib/tensor_forest/kernels/v4/fertile-stats-resource.h

```diff
@@ -69,6 +69,14 @@ class FertileStatsResource : public ResourceBase {
       const InputTarget* target, const std::vector<int>& examples,
       int32 node_id, bool* is_finished);
 
+  // Doing same thing as above but with pre-generated random features, so it's
+  // thread safe
+  void AddExampleToStatsAndInitialize(
```

Comment:
ditto, unless there is a clear usecase when u need the one from line 67, i would suggest just augementing the args of the original method. Besides, it looks like u always create this dict now (either with a provided seed or with a time stamp seed)

Reply:
there is a clear use case, which is default parallel option https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/tensor_forest/python/tensor_forest.py#L181  `collate_examples` = False